### PR TITLE
chore(ci): adds circle CI config file.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR adds a E2E basic test using the docker-compose.yml and the zipkin examples to create a trace and later retrieve it.

Ping @adriancole @JBAhire @kotharironak

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          name: Install Docker Compose
+          command: |
+            set -x
+            curl -L https://github.com/docker/compose/releases/download/1.24.1/docker-compose-`uname -s`-`uname -m` > ~/docker-compose
+            chmod +x ~/docker-compose
+            sudo mv ~/docker-compose /usr/local/bin/docker-compose
+
+      - run: jq --version
+      - run: docker-compose -f docker/docker-compose.yml up -d
+      - run: curl -i http://localhost:8081
```
Comment (adriancole) [inline .circleci/config.yml:21]: probably 2020 is better as that's the hypertrace UI port.. that or parse docker-compose for healthy.. that or put a TODO to expose a easy api to fetch a trace by ID that we upload

Comment (jcchavezs) [reply .circleci/config.yml:20]: This curl was incorrect. I am changing it now.

**Comment by adriancole** [comment]
fyi if you take out the filterBy here it is the same as counting traces I think. might be handy https://github.com/hypertrace/hypertrace-graphql#1-verify-trace-exists

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+jobs:
+  build:
+    docker:
+      - image: cimg/base:2020.01
+    working_directory: ~/hypertrace
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run: # official docs won't work, see https://github.com/circleci/circleci-docs/issues/1323#issuecomment-323620271
```
Comment (adriancole) [inline .circleci/config.yml:11]: and this is why we have to know too much to be ok at the DevOps

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+version: 2
+
+jobs:
+  build:
+    docker:
+      - image: cimg/base:2020.01
+    working_directory: ~/hypertrace
+    steps:
+      - checkout
+      - setup_remote_docker
```
Comment (adriancole) [inline .circleci/config.yml:10]: this is going to be very old docker btw. might want to indicate newer version:

```
      # The remote docker version is independent from what's installed in the gradle_docker image and defaults to Docker 17.
      #   Use the latest value from https://circleci.com/docs/2.0/building-docker-images/#docker-version
      - setup_remote_docker:
          version: 19.03.12
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      - run: # official docs won't work, see https://github.com/circleci/circleci-docs/issues/1323#issuecomment-323620271
+          name: Install Docker Compose
+          command: |
+            set -x
+            curl -L https://github.com/docker/compose/releases/download/1.24.1/docker-compose-`uname -s`-`uname -m` > ~/docker-compose
+            chmod +x ~/docker-compose
+            sudo mv ~/docker-compose /usr/local/bin/docker-compose
+
+      # we run docker-compose up but if it fails we print the logs for WARNING and ERROR
```
Comment (jcchavezs) [inline .circleci/config.yml:19]: no longer valid

File: docker/docker-compose.yml [hunk 5]
```diff
@@ -42,8 +49,10 @@ services:
         # note : To work successfully, it needs pinot to be healthy. This has started parallel for
         # startup experience.
         condition: service_started
+    links:
```
Comment (kotharironak) [inline docker/docker-compose.yml:52]: why do we need this?

File: .circleci/tests.sh [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+echo "Retrieving the list of traces:"
+echo ""
+
+TRACES=$(curl http://${TRACES_SERVER_HOST}:2020/graphql -H 'Content-Type: application/json' --data-binary \
+'{"query":"{
+  traces(
+    type: API_TRACE
+    between: {
+      startTime: \"2020-01-01T00:00:00.000Z\"
```
Comment (kotharironak) [inline .circleci/tests.sh:32]: interesting!! we can see later if we can pass this as an argument startTime, endTime?

File: docker/docker-compose.yml [hunk 7]
```diff
@@ -160,9 +177,19 @@ services:
...
+        "-h",
+        "hypertrace-federated-service",
+        "-p",
+        "9002",
+        "--timeout=120",
+        "--",
+        "start-bootstrapper.sh",
+      ]
+    healthcheck:
+      start_period: 180s
```
Comment (kotharironak) [inline docker/docker-compose.yml:192]: Is this needed?

File: .circleci/tests.sh (hunk 6 above)
Comment (jcchavezs) [reply .circleci/tests.sh:32]: I think so but I feel this is more like smoke test so probably not need to be that precise about time range.

File: docker/docker-compose.yml (hunk 5 above)
Comment (jcchavezs) [reply docker/docker-compose.yml:52]: While debugging I found in the pas that this service was complaining about zookeeper being unknown. Maybe due to docker version or something else. If I get this green then I will remove it.

File: docker/docker-compose.yml (hunk 7 above)
Comment (jcchavezs) [reply docker/docker-compose.yml:192]: Copied from your PR I think. I could remove it easily.

File: docker/docker-compose.yml [hunk 8]
```diff
@@ -160,9 +177,19 @@ services:
...
+        "wait-for-it.sh",
+        "-h",
+        "hypertrace-federated-service",
+        "-p",
+        "9002",
+        "--timeout=120",
+        "--",
+        "start-bootstrapper.sh",
+      ]
+    healthcheck:
```
Comment (jcchavezs) [inline docker/docker-compose.yml:191]:
```suggestion

```

File: .circleci/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,8 @@
+FROM alpine:3.7
```
Comment (adriancole) [inline .circleci/Dockerfile:1]: our other base images are 3.12 so this should reduce downloading I think..

```suggestion
FROM alpine:3.12
```

**Comment by adriancole** [comment]
@jcchavezs you may have better luck rebasing or merging against main per https://github.com/hypertrace/hypertrace/pull/37

**Comment by jcchavezs** [comment]
Something I see in my PR is that it is failing calling the `wait-for-it` script:

```
ERROR: for all-views-creator  Cannot start service all-views-creator: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"wait-for-it.sh\": executable file not found in $PATH": unknown

ERROR: for config-bootstrapper  Cannot start service config-bootstrapper: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"wait-for-it.sh\": executable file not found in $PATH": unknown
```

Logs: https://app.circleci.com/pipelines/github/hypertrace/hypertrace/80/workflows/d01bfa9d-2219-4277-957b-91313c95db0c/jobs/81

**Comment by adriancole** [comment]
you need to cd to the docker dir (vs reference the files from root). that
should fix the wait-for-it until we no longer use/need it

ex this is in circleci

docker-compose -f docker/docker-compose.yml -f
docker/docker-compose-zipkin-example.yml up -d || (docker-compose -f
docker/docker-compose.yml -f docker/docker-compose-zipkin-example.yml logs;
exit 1)

**Comment by jcchavezs** [comment]
@adriancole I am not entirely sure about the `cd docker`. Not only because it did not fix the problem to me but also because of docker documentation:

```
  --project-directory PATH    Specify an alternate working directory
                              (default: the path of the Compose file)
```

**Comment by adriancole** [comment]
I'm not sure either but I know that the volume mount is relative to docker/ so that is the only way I can currently explain this. Frankly circleci has been tricky to get working for anything see also the many overrides needed in remote docker!

**Review by JBAhire — APPROVED** [review]

File: docker/docker-compose.yml [hunk 10]
```diff
@@ -132,7 +133,14 @@ services:
     image: mvertes/alpine-mongo
     container_name: mongo
     healthcheck:
-      test: ["CMD", "mongo", "--quiet", "--eval", "'quit(db.runCommand({ ping: 1 }).ok ? 0 : 1)'"]
+      test:
```
Comment (adriancole) [inline docker/docker-compose.yml:136]: is this just formatting?

File: docker/docker-compose.yml (hunk 5 above)
Comment (adriancole) [reply docker/docker-compose.yml:30]: please do as it is green now!

**Review by JBAhire — DISMISSED** [review]

File: docker/docker-compose.yml (hunk 10 above)
Comment (jcchavezs) [reply docker/docker-compose.yml:136]: yes, somehow "vscode" does this and I could not disable it. Let me try again.

**Review by JBAhire — APPROVED** [review]

File: docker/docker-compose.yml (hunk 10 above)
Comment (adriancole) [reply docker/docker-compose.yml:134]: fine with me

**Review by adriancole — APPROVED** [review]
merge when ready!
